core/src/main/java/com/redhat/lightblue/client/request/data/GenerateRequest.java

```diff
@@ -0,0 +1,98 @@
+package com.redhat.lightblue.client.request.data;
+
+
+import java.util.List;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.redhat.lightblue.client.Operation;
+import com.redhat.lightblue.client.Projection;
+import com.redhat.lightblue.client.Query;
+import com.redhat.lightblue.client.Sort;
+import com.redhat.lightblue.client.http.HttpMethod;
+import com.redhat.lightblue.client.request.AbstractLightblueDataRequest;
+
+/**
+ * Generate a value using a field's value generator
+ *
+ * path: Path for the field whose value generator will be invoked
+ * n: Number of values to generate, defaults to 1
+ *
+ * Parse the results as:
+ * <pre>
+ *	String[] results = client.data(request, String[].class);
+ * </pre>
+ * 
+ */
+public class GenerateRequest extends AbstractLightblueDataRequest {
+
+    private Integer n;
+    private String path;
```

Comment:
Added null check.

On Wed, May 4, 2016 at 1:37 PM, Naveen Malik <notifications@github.com>
wrote:

> In
> core/src/main/java/com/redhat/lightblue/client/request/data/GenerateRequest.java
> <https://github.com/lightblue-platform/lightblue-client/pull/292#discussion_r62100780>
> :
>
> > +/**
> > + * Generate a value using a field's value generator
> > + *
> > + * path: Path for the field whose value generator will be invoked
> > + * n: Number of values to generate, defaults to 1
> > + *
> > + * Parse the results as:
> > + * <pre>
> > + *	String[] results = client.data(request, String[].class);
> > + * </pre>
> > + *
> > + */
> > +public class GenerateRequest extends AbstractLightblueDataRequest {
> > +
> > +    private Integer n;
> > +    private String path;
>
> Given null could be passed to constructor it's not going to fix anything.
> If anything have getRestURI fail if there's no path? Would save a request
> to lightblue if it was malformed.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-client/pull/292/files/85e4994abeffb4aebf1f052b533209bcd6b6c1e1#r62100780>
>